planner/optimize.go

```diff
@@ -176,6 +176,25 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 	for _, warn := range warns {
 		sessVars.StmtCtx.AppendWarning(warn)
 	}
+
+	defer func() {
+		// Override the resource group if necessary
+		// TODO: we didn't check the existence of the hinted resource group now to save the cost per query
+		if retErr == nil && sessVars.StmtCtx.StmtHints.HasResourceGroup {
+			if variable.EnableResourceControl.Load() {
+				sessVars.ResourceGroupName = sessVars.StmtCtx.StmtHints.ResourceGroup
```

Comment:
Is the hint statement level, thus it needs to be reset after the statement finishes in a transaction?

Reply:
Yes, actually we do reset the `sessVars.ResourceGroupName` after this statement finish executing.